### DISCUSSION TITLE
Is it possible to use native mobile libraries

### DISCUSSION THREAD

**Discussion by trydalch** [unoplatform/uno#18918] [Q&A] [answered]
In React Native, you can write native code or use native libraries and link it to your React Native app. Is it possible to achieve this with Uno Platform? 

I'd really like to be able to use tools like [Superwall](https://superwall.com/) but they don't have a .net SDK, though they do have Flutter and ReactNative SDKs which just wrap the native calls. 

Edit: It looks like it's [supported in Maui](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/native-library-interop/get-started), though it's questionable whether it supports libraries written in Swift. Additionally, while Maui controls can be embedded in Uno apps, I'm not sure if that makes all Maui features available.

**Comment by trydalch** [comment, marked as answer]
It looks like it's possible if a .NET Binding project/library is created. 

https://github.com/CommunityToolkit/maui.nativelibraryinterop

Then it can be used in our .net for ios and .net for android apps.

**Reply by jeromelaban** [reply]
Uno sits on top of .NET iOS/Android, you ca use [all the information available here](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/native-library-interop/get-started) as it is application to Uno as well.
